pkg/virt-launcher/virtwrap/agent-poller/agent_poller.go

```diff
@@ -341,72 +342,158 @@ func (p *AgentPoller) Stop() {
-			continue
-		}
-
 		// parse the json data and convert to domain api
 		// for libvirt 5.6.0 json conversion deprecated
 		switch command {
 		case GET_INTERFACES:
-			interfaces, err := parseInterfaces(cmdResult)
-			if err != nil {
-				log.Log.Errorf("Cannot parse guest agent interface %s", err.Error())
-				continue
+			if guestInfo, err := getGuestInfo(libvirt.DOMAIN_GUEST_INFO_INTERFACES, con, domainName); err == nil {
+				agentStore.Store(GET_INTERFACES, convertToInterfaces(guestInfo))
 			}
-			agentStore.Store(GET_INTERFACES, interfaces)
 		case GET_OSINFO:
-			osInfo, err := parseGuestOSInfo(cmdResult)
-			if err != nil {
-				log.Log.Errorf("Cannot parse guest agent guestosinfo %s", err.Error())
-				continue
+			if guestInfo, err := getGuestInfo(libvirt.DOMAIN_GUEST_INFO_OS, con, domainName); err == nil {
+				agentStore.Store(GET_OSINFO, convertToOSInfo(guestInfo))
 			}
-			agentStore.Store(GET_OSINFO, osInfo)
 		case GET_HOSTNAME:
-			hostname, err := parseHostname(cmdResult)
-			if err != nil {
-				log.Log.Errorf("Cannot parse guest agent hostname %s", err.Error())
-				continue
+			if guestInfo, err := getGuestInfo(libvirt.DOMAIN_GUEST_INFO_HOSTNAME, con, domainName); err == nil {
+				agentStore.Store(GET_HOSTNAME, guestInfo.Hostname)
 			}
-			agentStore.Store(GET_HOSTNAME, hostname)
 		case GET_TIMEZONE:
-			timezone, err := parseTimezone(cmdResult)
-			if err != nil {
-				log.Log.Errorf("Cannot parse guest agent timezone %s", err.Error())
-				continue
+			if guestInfo, err := getGuestInfo(libvirt.DOMAIN_GUEST_INFO_TIMEZONE, con, domainName); err == nil {
+				agentStore.Store(GET_TIMEZONE, convertToTimezone(guestInfo))
 			}
-			agentStore.Store(GET_TIMEZONE, timezone)
 		case GET_USERS:
-			users, err := parseUsers(cmdResult)
+			if guestInfo, err := getGuestInfo(libvirt.DOMAIN_GUEST_INFO_USERS, con, domainName); err == nil {
+				agentStore.Store(GET_USERS, convertToUsers(guestInfo))
+			}
+		case GET_FSFREEZE_STATUS:
+			// TODO: Replace with direct call to libvirt function
+			cmdResult, err := con.QemuAgentCommand(`{"execute":"`+string(command)+`"}`, domainName)
 			if err != nil {
-				log.Log.Errorf("Cannot parse guest agent users %s", err.Error())
 				continue
 			}
-			agentStore.Store(GET_USERS, users)
-		case GET_FSFREEZE_STATUS:
+
 			fsfreezeStatus, err := ParseFSFreezeStatus(cmdResult)
 			if err != nil {
 				log.Log.Errorf("Cannot parse guest agent fsfreeze status %s", err.Error())
 				continue
 			}
+
 			agentStore.Store(GET_FSFREEZE_STATUS, fsfreezeStatus)
 		case GET_FILESYSTEM:
-			filesystems, err := parseFilesystem(cmdResult)
+			if guestInfo, err := getGuestInfo(libvirt.DOMAIN_GUEST_INFO_FILESYSTEM, con, domainName); err == nil {
+				agentStore.Store(GET_FILESYSTEM, convertToFileSystem(guestInfo))
+			}
+		case GET_AGENT:
+			// TODO: Replace with direct call to libvirt function
+			cmdResult, err := con.QemuAgentCommand(`{"execute":"`+string(command)+`"}`, domainName)
```

Comment:
There is a question why `guest-info` is used in the project itself, according to @berrange this command shouldn't be used by KubeVirt, a quote from his words:
```
the 'guest-info'  command shouldn't be relevant to kubevirt -
its just a low level way to query the agent version & supported commands.
so surprised kubevirt is using that.
```